File: src/main/java/io/xol/chunkstories/api/exceptions/UnknownComponentException.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.exceptions;

import io.xol.chunkstories.api.entity.Entity;

public class UnknownComponentException extends PacketProcessingException
{
	String message;
	
	public UnknownComponentException(int componentId, Class<? extends Entity> entityClass)
	{
		message = "The componentId : "+componentId+" for the entity "+entityClass.getName()+" was not found";
	}

	public UnknownComponentException(String componentName, Class<? extends Entity> entityClass)
	{
		message = "The componentId : "+componentName+" for the entity "+entityClass.getName()+" was not found";
	}

	@Override
	public String getMessage()
	{
		return message;
	}
	
	/**
	 * 
	 */
	private static final long serialVersionUID = -3592430343334562201L;

}

File: src/main/java/xyz/chunkstories/api/exceptions/UnknownComponentException.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.exceptions

import xyz.chunkstories.api.entity.Entity

class UnknownComponentException(override val message: String) : PacketProcessingException() {

    constructor(componentId: Int, entityClass: Class<out Entity>) : this(
            message = "The componentId : " + componentId + " for the entity " + entityClass.name + " was not found"
    )

    constructor(componentName: String, entityClass: Class<out Entity>) : this(
            message = "The componentId : " + componentName + " for the entity " + entityClass.name + " was not found"
    )

    companion object {
        private val serialVersionUID = -3592430343334562201L
    }
}
